omod/src/main/webapp/pages/labtrackingAddOrder.gsp

```diff
@@ -76,33 +77,29 @@ ${ ui.includeFragment("coreapps", "patientHeader", [ patient: patient ]) }
             <div class="form-group row">
                <label class="col-sm-2 col-form-label">${ui.message("labtrackingapp.prelabdiagnosislabel")}</label>
                <div class="col-sm-9">
-                  <input type="text" ng-model="order.preLabDiagnosis"  class="form-control" ng-required="false"
+                  <input ng-if="!order.uuid" type="text" ng-model="order.preLabDiagnosis"  class="form-control" ng-required="false"
                      uib-typeahead="item as item.label for item in alldiagnoses | filter:{label:${'$viewValue'}} | limitTo:8" />
+                  <p ng-if="order.uuid" class="form-control-static" > {{ formatDiagnosis(order.preLabDiagnosis) }}</p>
                </div>
             </div>
             <div class="form-group row">
                <label for="procedure" class="col-sm-2 col-form-label">${ui.message("labtrackingapp.proceduresitelabel")}</label>
                <div class="col-sm-4">
-                  <div class="panel panel-default">
-                     <div class="panel-heading">${ui.message("labtrackingapp.proceduresAvailable")}</div>
-                     <div class="panel-body">
-                        <select id="procedure" class="form-control" multiple
-                           ng-options="item as item.label for item in procedures | orderBy:'label' track by item.value "
-                           ng-model="selectedProcedures" ></select>
-                     </div>
-                  </div>
+                  <input  type="text" ng-model="selectedProcedures"  class="form-control" ng-required="false"
+                          uib-typeahead="item as item.label for item in procedures | filter:${'$viewValue'} | limitTo:8"
+                          typeahead-on-select="onSelectProcedure(${'$item'}, ${'model'}, ${'label'})"/>
                </div>
                <div class="col-sm-1">
                   <p>&nbsp;</p>
-                  <button class="btn button-wrapper" type="button" ng-disabled="order.procedures.length > 2" ng-click="addProcedure()"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span></button>
-                  <button class="btn button-wrapper" type="button" ng-disabled="order.procedures.length == 0" ng-click="removeProcedure()"><span class="glyphicon glyphicon-menu-left" aria-hidden="true"></span></button>
+                  <button class="btn button-wrapper" type="button" ng-disabled="order.proceduresForSpecimen.length > 2" ng-click="addProcedure()"><span class="glyphicon glyphicon-menu-right" aria-hidden="true"></span></button>
```

Comment:
Not a big deal, but are renaming this "proceduresForSpecimen" because there are other procedures captured until "order.procedures" elsewhere?

Reply:
The order.procedures are the procedures that were ordered and they are stored on the Order object. 
proceduresForSpecimen are the procedures performed and they are stores as obs on the Specimen Collection Encounter.

Reply:
In my testing, the concept_id on the order object is always a generic concept "Pathology procedure ordered" (which was by design, if I recall correctly).  Are the actual specific procedures stored somewhere else on the Order?